backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java

```diff
@@ -97,8 +102,12 @@ private static String getCaloriesFromJson(JsonObject jsonObject) {
   /* Structure of jsonObject:
    * {suitableForDiet: "http://schema.org/VegetarianDiet, http://schema.org/GlutenFreeDiet .."} */
   private static String[] getDietFromJson(JsonObject jsonObject) {
-    String dietElements = jsonObject.get("suitableForDiet").getAsString();
-    String[] diet = dietElements.split(", ");
+    String[] NO_DIET = {};
+    JsonElement dietElements = jsonObject.get("suitableForDiet");
+    if (dietElements == null) {
+      return NO_DIET;
+    }
```

Comment:
```suggestion
    JsonElement dietElements = jsonObject.get("suitableForDiet");
    if (dietElements == null) {
      return new String[0];
    }
```


Instead of creating a variable `NO_DIET` (wrong naming convention for local variables) you can just return an empty array